internal/collector/otel_collector_plugin.go

```diff
@@ -206,10 +206,35 @@ func (oc *Collector) Subscriptions() []string {
 	return []string{
 		bus.ResourceUpdateTopic,
 		bus.NginxConfigUpdateTopic,
-		bus.AgentConfigUpdateTopic,
 	}
 }
 
+func (oc *Collector) Reconfigure(ctx context.Context, agentConfig *config.Config) error {
+	slog.DebugContext(ctx, "OTel collector plugin received agent config update message")
+
+	oc.agentConfigMutex.Lock()
+	defer oc.agentConfigMutex.Unlock()
+
+	if oc.config.Collector != nil && oc.config.Collector.Extensions.HeadersSetter != nil &&
+		oc.config.Collector.Extensions.HeadersSetter.Headers != nil {
+		if !reflect.DeepEqual(oc.config.Collector.Extensions.HeadersSetter.Headers,
+			agentConfig.Collector.Extensions.HeadersSetter.Headers) {
+			slog.InfoContext(ctx, "OTel collector headers have changed, restarting collector")
+			oc.config = agentConfig
+			oc.restartMutex.Lock()
+			defer oc.restartMutex.Unlock()
+			oc.restartCollector(ctx)
+		}
+	} else {
+		oc.config = agentConfig
```

Comment:
Can you add an info log here saying that new headers have being added and the collector is bing restarted